### PR TITLE
docs(contrib): Document things I look for in RFCs

### PR DESCRIPTION
**Pull request by epage**
### What does this PR try to resolve?

Trying to make the RFC experience more pleasant by being upfront about things they need to take into account, rather than waiting for Cargo team members to review it (which they can also miss considering these points).

Along the way, I ended up tweaking the unstable docs as they are a related part of the process.

### CODE REVIEW

File: src/doc/contrib/src/process/rfc.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+For example,
+- [RFC 3537] chose
+  configuration for the MSRV-aware resolver because users would likely need
+  to change this setting, like in CI to verify the opposite case of
+  what they run by default.
+- The Cargo team rejected a [`[cfg]` table][cfg table] to represent `rustc`
+  `--cfg` flags as it was a direct port of low-level rustc behavior that didn't
+  mesh with the other high level abstractions of manifests.
+  - For stabilization, this was worked around through a build script directive and a `[lints]` field configuration.
+- [#12738][cargo#12738] for exploring how existing config might be representabl in `Cargo.toml`.
```
Comment (weihanglo) [inline src/doc/contrib/src/process/rfc.md:29]:
```suggestion
- [#12738][cargo#12738] for exploring how existing config might be representable in `Cargo.toml`.
```

File: src/doc/contrib/src/process/rfc.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+- How does specifying a field in both `workspace` and a package interact?
+  - e.g. dependency sources cannot be overridden
+  - e.g. dependency `features` get merged
+  - e.g. depedency `default-features` has been hard to get right ([#12162][cargo#12162])
+
+[RFC 3389]: https://rust-lang.github.io/rfcs/3389-manifest-lint.html
+[cargo#12162]: https://github.com/rust-lang/cargo/issues/12162
+
+#### `dependencies`
+
```
Comment (weihanglo) [inline src/doc/contrib/src/process/rfc.md:62]: We don't use new heading for other places. T make it more consistent, maybe this?

```suggestion
When working on `dependencies` tables,
```

File: src/doc/contrib/src/process/rfc.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
+# Writing an RFC
+
+Generally, an RFC goes through:
+1. Pre-RFC discussions on [irlo]
```
Comment (weihanglo) [inline src/doc/contrib/src/process/rfc.md:4]: Not everyone recognizes the abbreviation.

```suggestion
1. Pre-RFC discussions on [the internal fourm (IRLO)][irlo]
```

File: src/doc/contrib/src/process/unstable.md [hunk 4]
```diff
@@ -37,7 +37,9 @@ something is only partially implemented, it may have both
 [S-accepted]: https://github.com/rust-lang/cargo/labels/S-accepted
 [S-waiting-on-feedback]: https://github.com/rust-lang/cargo/labels/S-waiting-on-feedback
 
-## Unstable feature opt-in
+## Implementation
+
+#### Unstable feature opt-in
```
[the remark this answers is not in the document]
Comment (epage) [reply src/doc/contrib/src/process/unstable.md:50]: Yes, when browsing the site, h3 didn't seem to have enough differentiation from h2

**Review by weihanglo — APPROVED** [review]

**Comment by weihanglo** [comment]
Thanks for the write-up!

@bors r+

**Pull request merged**